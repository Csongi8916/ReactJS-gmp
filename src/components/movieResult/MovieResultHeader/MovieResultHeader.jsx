import React from 'react';
import NxrToogleButton from '../../UI/NxrToogleButton/NxrToogleButton';

import classes from './MovieResultHeader.scss';


export default function MovieResultHeader(props) {
  return (
    <div className="MovieResultHeader">
      <div className="MovieResultHeader_LeftInfoPanel">
        {props.needTotal && <span>7 movies found</span>}
      </div>
      <div className="MovieResultHeader_RightInfoPanel">
        <NxrToogleButton title="Sort by" firstTitle="released date" secondTitle="raiting" />
      </div>
    </div>
  )
}
