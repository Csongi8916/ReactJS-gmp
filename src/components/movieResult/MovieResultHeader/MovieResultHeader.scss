$nrRedColor: #f65261;
$nrBlackColor: #232323;
$nrGrayColor: #424242;
$nrBgColor: #555555;


.MovieResultHeader {
  display: flex;
  align-items: center;
  height: 60px;
  background-color: $nrBgColor;

  .MovieResultHeader_LeftInfoPanel {
    flex-grow: 1;
    margin-left: 50px;
  }

  .MovieResultHeader_RightInfoPanel {
    display: flex;
    align-items: center;
    margin-right: 50px;
  }
}