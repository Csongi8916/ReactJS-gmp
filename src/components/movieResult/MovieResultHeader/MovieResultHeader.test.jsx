import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import MovieResultHeader from './MovieResultHeader';

describe('MovieResultHeader Snapshpt tests', () => {
  test('Should render correctly MovieResultHeader', () => {
    const tree = renderer.create(<MovieResultHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MovieResultHeader UI tests', () => {
  it('Should number of movie equal header title info', () => {
    const wrap = shallow(<MovieResultHeader needTotal={true} count={5} />);
    const count = wrap.text().search('5');
    expect(count).toBeDefined();
  });
});
