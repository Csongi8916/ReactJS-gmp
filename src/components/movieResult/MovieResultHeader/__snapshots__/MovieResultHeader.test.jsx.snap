// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieResultHeader Snapshpt tests Should render correctly MovieResultHeader 1`] = `
<div
  className="MovieResultHeader"
>
  <div
    className="MovieResultHeader_LeftInfoPanel"
  >
    <span>
      Films by drama genre
    </span>
  </div>
  <div
    className="MovieResultHeader_RightInfoPanel"
  />
</div>
`;
